diesel_derives/src/insertable.rs

```diff
@@ -85,23 +62,23 @@ pub fn derive(item: syn::DeriveInput) -> Result<proc_macro2::TokenStream, Diagno
                 direct_field_ty.push(field_ty_serialize_as(
                     field,
                     table_name,
-                    &ty,
+                    ty,
                     treat_none_as_default_value,
                 ));
                 direct_field_assign.push(field_expr_serialize_as(
                     field,
                     table_name,
-                    &ty,
+                    ty,
                     treat_none_as_default_value,
                 ));
 
                 generate_borrowed_insert = false; // as soon as we hit one field with #[diesel(serialize_as)] there is no point in generating the impl of Insertable for borrowed structs
             }
-            (Some(_), true) => {
-                return Err(field
-                    .flags
-                    .span()
-                    .error("`#[diesel(embed)]` cannot be combined with `#[diesel(serialize_as)]`"))
+            (Some(ty), true) => {
+                abort!(
+                    ty,
```

Comment:
Shouldn't that point to the complete field or even to the derive itself instead of just the type?

Reply:
Agreed, I changed this to put to the span of the field. But I think there's a bug somewhere (either in syn or rust).

```
error: `#[diesel(embed)]` cannot be combined with `#[diesel(serialize_as)]`
  --> $DIR/embed_and_serialize_as_cannot_be_mixed.rs:22:5
   |
22 |     #[diesel(embed, serialize_as = SomeType)]
   |     ^
```

The span just points to the beginning of the derive and nothing else. Shall I leave it as that?

Reply:
Yeah, the above error message you see is what's happening when I use span of the field.

Reply:
I will try to have a look at that this week. Otherwise thats something I would not consider to be a blocking issue for merging this PR.

Reply:
Okay, you can check this once I update the PR then.

Reply:
I've pushed a fixed version. I've also fixed some copy paste error in a `#[derive(QueryableByName)]` error message.